### PR TITLE
fix #512

### LINKED ISSUE #512: VisualDL不支持paddle模型的展示
VisualDL不支持paddle模型的展示

### PR DESCRIPTION
**Pull request by nepeplwu**
support show paddle model，user can use **paddle.fluid.io.save_inference_model()** to save their model to <SAVE_MODEL_PATH>, and then use **visualdl --model_pb <SAVE_MODEL_PATH>** to load the model

### CODE REVIEW

File: frontend/src/graph/ui/Chart.vue [hunk 1]
```diff
@@ -176,41 +158,134 @@ export default {
...
+            }
+            let arr = dic[inputData]['output'];
+            arr[arr.length] = nodeKey;
+          }
+        }
+      }
 
+      for (let i in dic) {
+        if (dic.hasOwnProperty(i)) {
+          for (let input in dic[i]['input']) {
```
Comment (Superjomn) [inline frontend/src/graph/ui/Chart.vue]: so deep nest ...

File: frontend/src/graph/ui/Chart.vue [hunk 2]
```diff
@@ -176,41 +158,134 @@ export default {
...
-          for (let e = 0; e < curOperatorNode['input'].length; ++e) {
-            g.setEdge(curOperatorNode['input'][e], nodeKey);
+      // add input nodes
+      if (has(graphData, 'input') === true && drawInputNode === true) {
+        for (let i = 0; i < graphData['input'].length; ++i) {
+          let name = graphData['input'][i]['name'];
+          inputMap[name] = i;
+        }
+        console.log(inputMap);
+
```
Comment (Superjomn) [inline frontend/src/graph/ui/Chart.vue]: split to a function call? make the code shorter please.

File: visualdl/server/model/paddle/paddle2graph.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                cnt = var_count_dict[argument]
+                val = "%s@%d" % (argument, cnt)
+                self.outputs[argument] = val
+
+    def set_input_name(self, key, name):
+
+        self.inputs[key] = name
+
+    def set_output_name(self, key, name):
+
```
Comment (Superjomn) [inline visualdl/server/model/paddle/paddle2graph.py]: is the empty line necessary for a one-line method?

File: visualdl/server/model/paddle/paddle2graph.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        return self.input_ops.difference(self.output_ops)
+
+    def get_info(self):
+
+        return self.info
+
+
+class PaddleModel:
+
```
Comment (Superjomn) [inline visualdl/server/model/paddle/paddle2graph.py]: no empty line here, add some comment instead.

File: visualdl/server/model/paddle/paddle2graph.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        # remove vars bind with fetch operator
+        for key, op in self.ops.items():
+            if op.is_fetch_op():
+                self.output_set = self.output_set.union(op.get_input_set())
+                self.input_set = self.input_set.difference(op.get_input_set())
+                self.output_set = self.output_set.difference(
+                    op.get_output_set())
+
+        for input in self.input_set:
+
```
Comment (Superjomn) [inline visualdl/server/model/paddle/paddle2graph.py]: no empty line here, make the code compact.

File: visualdl/server/visualdl [hunk 6]
```diff
@@ -319,7 +320,14 @@ def individual_audio():
 
 @app.route('/data/plugin/graphs/graph')
 def graph():
-    json_str = vdl_graph.draw_graph(args.model_pb)
+
```
Comment (Superjomn) [inline visualdl/server/visualdl]: the empty line in the head of a method definition is not needed!

please change the code across the whole PR.

**Review by Superjomn — CHANGES_REQUESTED** [review]
Some code style issue, please add more comment, and make the code more compact.

Some if-else takes too many levels of nests, please consider include some functions and comment to make the code more readable.

File: visualdl/server/model/paddle/framework.proto [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+syntax = "proto2";
+option optimize_for = LITE_RUNTIME;
+package paddle.framework.proto;
+
+// Any incompatible changes to ProgramDesc and its dependencies should
```
Comment (Superjomn) [inline visualdl/server/model/paddle/framework.proto:21]: Add some comments to indicate the source of this proto file, such as

`This file is borrowed from project PaddlePaddle to better support the models from it.`

**Review by Superjomn — APPROVED** [review]
LGTM

File: frontend/src/graph/ui/Chart.vue (hunk 2 above)
Comment (nepeplwu) [reply frontend/src/graph/ui/Chart.vue]: thanks for review, i haved split the code into some function call

File: visualdl/server/model/paddle/paddle2graph.py (hunk 3 above)
Comment (nepeplwu) [reply visualdl/server/model/paddle/paddle2graph.py]: thanks for review, i haved removed the empty line in the new version

File: visualdl/server/model/paddle/paddle2graph.py (hunk 4 above)
Comment (nepeplwu) [reply visualdl/server/model/paddle/paddle2graph.py]: thanks for review, i haved removed the empty line in the new version

File: visualdl/server/model/paddle/paddle2graph.py (hunk 5 above)
Comment (nepeplwu) [reply visualdl/server/model/paddle/paddle2graph.py]: thanks for review, i haved removed the empty line in the new version

File: visualdl/server/visualdl (hunk 6 above)
Comment (nepeplwu) [reply visualdl/server/visualdl]: thanks for review, i haved removed the empty line in the new version

File: visualdl/server/model/paddle/framework.proto (hunk 7 above)
Comment (nepeplwu) [reply visualdl/server/model/paddle/framework.proto:21]: thanks for review, i haved add note in new version

File: frontend/src/graph/ui/Chart.vue (hunk 1 above)
Comment (nepeplwu) [reply frontend/src/graph/ui/Chart.vue]: thanks for review, i  modified the call logic of the if statement

**Comment by nickyfantasy** [comment]
LGTM

**Review by nickyfantasy — APPROVED** [review]

**Pull request merged**